catalog/ui/search-ui/standard/src/main/webapp/js/widgets/cesium.mapclustering.js

```diff
@@ -41,53 +41,82 @@ define(['marionette',
+                            // hide entity
                             this.hideEntity(billboard, view);
                         }
                     }
                 }, this);
 
+                // Remove old entities
                 _.each(this.clusters, function(value) {
-                    if(value.points.length > 1) {
-                        var newCluster = {
-                            position : value.position,
-                            point : {
-                                pixelSize : 25,
-                                color : MARKER_COLOR
-                            },
-                            label : {
-                                text : "" + value.points.length,
-                                font : '14px Helvetica',
-                                style : Cesium.LabelStyle.FILL
-                            }
-                        };
-                        value.entity = this.viewer.entities.add(newCluster);
+                    var cluster = this.getCluster(value, clusters);
+                    if(_.isUndefined(cluster)) {
+                        this.viewer.entities.remove(value.entity);
+                    }
+                }, this);
+
+                _.each(clusters, function(value) {
+                    var cluster = this.getCluster(value, this.clusters);
+                    // Render new entities
+                    if(_.isUndefined(cluster)) {
+                        if(value.points.length > 1) {
+                            var newCluster = {
+                                position : value.position,
+                                point : {
+                                    pixelSize : 25,
+                                    color : MARKER_COLOR
+                                },
+                                label : {
+                                    text : "" + value.points.length,
+                                    font : '14px Helvetica',
+                                    style : Cesium.LabelStyle.FILL
+                                }
+                            };
+                            value.entity = this.viewer.entities.add(newCluster);
+                        } else {
+                            this.showEntity(value.ref, value.views[0]);
+                        }
                     } else {
-                        this.showEntity(value.ref, value.views[0]);
+                        // Set existing entities in the new cluster array and show non clustered points
+                        value.entity = cluster.entity;
+                        if(cluster.points.length === 1) {
+                            this.showEntity(cluster.ref, cluster.views[0]);
+                        }
                     }
                 }, this);
+                this.clusters = clusters;
             }
         },
+        getCluster : function(cluster, clusters) {
+             var result;
+            _.each(clusters, function(value) {
```

Comment:
I think this would work:
```
var result = _.find(clusters, function(value) {
    return _.isEqual(value.views, cluster.views);
});
return result; // or just directly return the result of the above function
```